src/snowflake/snowpark/functions.py

```diff
@@ -3019,28 +3220,112 @@ def array_to_string(array: ColumnOrName, separator: ColumnOrName) -> Column:
     return builtin("object_agg")(k, v)
 
 
 def object_construct(*key_values: ColumnOrName) -> Column:
-    """Returns an OBJECT constructed from the arguments."""
+    """Returns an OBJECT constructed from the arguments.
+
+    Example::
+
+        >>> from snowflake.snowpark.types import StructType, StructField, VariantType, StringType
+        >>> df = session.create_dataframe(
+        ...     [["name", "Joe"], ["zip", "98004"]],
+        ...     schema=StructType([StructField("k", StringType()), StructField("v", VariantType())])
+        ... )
+        >>> df.select(object_construct(col("k"), col("v")).alias("result")).show()
+        --------------------
+        |"RESULT"          |
+        --------------------
+        |{                 |
+        |  "name": "Joe"   |
+        |}                 |
+        |{                 |
+        |  "zip": "98004"  |
+        |}                 |
+        --------------------
+        <BLANKLINE>
+    """
     kvs = [_to_col_if_str(kv, "object_construct") for kv in key_values]
     return builtin("object_construct")(*kvs)
 
 
 def object_construct_keep_null(*key_values: ColumnOrName) -> Column:
     """Returns an object containing the contents of the input (i.e. source) object with one or more
-    keys removed."""
+    keys removed.
+
+    Example::
+
+        >>> from snowflake.snowpark.types import StructType, StructField, VariantType, StringType
+        >>> df = session.create_dataframe(
+        ...     [["key_1", "one"], ["key_1", None]],
+        ...     schema=StructType([StructField("k", StringType()), StructField("v", VariantType())])
+        ... )
+        >>> df.select(object_construct_keep_null(col("k"), col("v")).alias("result")).show()
+        --------------------
+        |"RESULT"          |
+        --------------------
+        |{                 |
+        |  "key_1": "one"  |
+        |}                 |
+        |{                 |
+        |  "key_1": null   |
+        |}                 |
+        --------------------
+        <BLANKLINE>
+    """
     kvs = [_to_col_if_str(kv, "object_construct_keep_null") for kv in key_values]
     return builtin("object_construct_keep_null")(*kvs)
 
 
 def object_delete(obj: ColumnOrName, key1: ColumnOrName, *keys: ColumnOrName) -> Column:
-    """Returns an object consisting of the input object with a new key-value pair inserted.
-    The input key must not exist in the object."""
+    """Returns an object consisting of the input object with one or more keys removed.
+    The input key must not exist in the object.
+
+    Example::
+
+        >>> df = session.sql(
+        ...     "select object_construct(a,b,c,d,e,f) as obj, k, v from "
+        ...     "values('age', 21, 'zip', 21021, 'name', 'Joe', 'age', 0),"
+        ...     "('age', 26, 'zip', 94021, 'name', 'Jay', 'age', 0) as T(a,b,c,d,e,f,k,v)"
+        ... )
+        >>> df.select(object_delete(col("obj"), col("k")).alias("result")).show()
```

Comment:
There seems to be no keys removed here? Could we show a case where `keys` is not empty?

Reply:
I've updated the sample code to make it explicitly that key "age" is removed